packages/foam-core/src/core.ts

```diff
@@ -0,0 +1,89 @@
+import { Graph, Edge } from 'graphlib'
+
+
+type ID = string
+
+export interface Link {
+  from: ID
+  to: ID
+  text: string
+}
+
+export interface NoteLink {
+  to: ID
+  text: string
+}
+
+
+export class Note {
+  public id: ID
+  public title: string
+  public source: string
+  public path: string
+  public links: NoteLink[]
+
+  constructor(id: ID, title: string, links: NoteLink[], path: string, source: string) {
+    this.id = id
+    this.title = title
+    this.source = source
+    this.path = path
+    this.links = links
+  }
+}
+
+
+export class NoteGraph {
+  private graph: Graph
```

Comment:
Should this contain other fields like https://github.com/foambubble/foam/blob/master/packages/foam-workspace-manager/src/WorkspaceManager.ts#L50

Reply:
Currently the old WorkspaceManager.path was never used, but it may be useful considering that in the graph nodes we want to keep relative urls, and it may be useful to have the information at hand to construct absolute file URIs

Reply:
yeah I removed it because it wasn't used, and I was debating whether this info should live in the config or the graph. currently we store the absolute paths in the graph, having a base path would allow us to make those relative. why do you prefer relative urls @jevakallio ?

Reply:
@riccardoferretti that's a good question. 

I guess I prefer them for readability -- the paths will be closer to what's being used in links and generated into link ref defs -- but that's not a strong preference.

Another argument for relative paths is that if we ever persist the graph, relative links continue to work when e.g. cloning the repo to different computer, moving it to different folders, or even as logical paths in a GitHub repo / published site.

If there's no good reason for absolute paths in `Note.path`, I'd rather keep it relative, but can be convinced otherwise.